src/model/kubernetes.js

```diff
@@ -0,0 +1,329 @@
+                      ],
+                    },
+                  },
+                },
+              },
+            ],
+            restartPolicy: 'Never',
+          },
+        },
+        backoffLimit: 1,
+      },
+    };
+    await this.kubeClient.apis.batch.v1.namespaces('default').jobs.post({ body: jobManifest });
+    core.info('Job created');
+  }
+
+  static async watchBuildJobUntilFinished() {
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+
+    let podname;
+    let ready = false;
+    while (!ready) {
+      // eslint-disable-next-line no-await-in-loop
+      const pods = await this.kubeClient.api.v1.namespaces('default').pods.get();
+      // eslint-disable-next-line no-plusplus
+      for (let index = 0; index < pods.body.items.length; index++) {
+        const element = pods.body.items[index];
+        if (element.metadata.labels['job-name'] === this.jobName) {
+          if (element.status.phase !== 'Pending') {
+            core.info('Pod no longer pending');
+            if (element.status.phase === 'Failure') {
+              core.error('Kubernetes job failed');
+            } else {
+              ready = true;
+              podname = element.metadata.name;
+            }
+          }
+        }
+      }
+    }
+
+    if (!podname) {
+      core.error('no pods to stream logs from found');
+    }
+
+    core.info(`Watching build job ${podname}`);
+    let logQueryTime;
+    let complete = false;
+    while (!complete) {
+      // eslint-disable-next-line no-await-in-loop
+      const podStatus = await this.kubeClient.api.v1.namespaces('default').pod(podname).get();
+      if (podStatus.body.status.phase !== 'Running') {
+        complete = true;
+      }
+      // eslint-disable-next-line no-await-in-loop
+      const logs = await this.kubeClient.api.v1
+        .namespaces('default')
+        .pod(podname)
+        .log.get({
+          qs: {
+            sinceTime: logQueryTime,
+            timestamps: true,
+          },
+        });
+      if (logs.body !== undefined) {
+        const arrayOfLines = logs.body.match(/[^\n\r]+/g).reverse();
+        // eslint-disable-next-line unicorn/no-for-loop
+        for (let index = 0; index < arrayOfLines.length; index += 1) {
+          const element = arrayOfLines[index];
+          const [time, ...line] = element.split(' ');
+          if (time !== logQueryTime) {
+            core.info(line.join(' '));
+          } else {
+            break;
+          }
+        }
+
+        if (podStatus.body.status.phase === 'Failed') {
+          throw new Error('Kubernetes job failed');
+        }
+
+        // eslint-disable-next-line prefer-destructuring
+        logQueryTime = arrayOfLines[0].split(' ')[0];
+      }
+    }
+  }
+
+  static async cleanupJob() {
+    await this.kubeClient.apis.batch.v1.namespaces('default').jobs(this.jobName).delete();
+    await this.kubeClient.api.v1.namespaces('default').secrets(this.secretName).delete();
+  }
```

Comment:
Suggest naming this `cleanup` as it seems to be the general method for any cleanup and could later be split into multiple methods if needed.